### ISSUE TITLE
[CI] ProxyConnectionStrategyTests.testProxyStrategyWillOpenNewConnectionsOnDisconnect failures

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#50618]
This started failing recently on master and 7.x occasionally. Latest failure:
Log | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=openjdk13,ES_RUNTIME_JAVA=adoptopenjdk11,nodes=general-purpose/427/console
-- | --
Build Scans | [7.5.2] https://gradle-enterprise.elastic.co/s/m5zn7f4jy7dwu [7.6.0] https://gradle-enterprise.elastic.co/s/d3prskq3yejbchttps://gradle-enterprise.elastic.co/s/zooioeyhqehsu

Does not reproduce locally for me
```
./gradlew ':server:test' --tests "org.elasticsearch.transport.ProxyConnectionStrategyTests.testProxyStrategyWillOpenNewConnectionsOnDisconnect" \
  -Dtests.seed=5ACE63EDC9EA6CE8 \
  -Dtests.security.manager=true \
  -Dtests.locale=bg-BG \
  -Dtests.timezone=Africa/Kinshasa \
  -Dcompiler.java=13
```

```
java.lang.AssertionError: expected:<8> but was:<7>
	at __randomizedtesting.SeedInfo.seed([5ACE63EDC9EA6CE8:B7C7088639FC7A80]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.elasticsearch.transport.ProxyConnectionStrategyTests.lambda$testProxyStrategyWillOpenNewConnectionsOnDisconnect$8(ProxyConnectionStrategyTests.java:152)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:879)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:852)
	at org.elasticsearch.transport.ProxyConnectionStrategyTests.testProxyStrategyWillOpenNewConnectionsOnDisconnect(ProxyConnectionStrategyTests.java:145)
```

**Comment by cbuescher** [comment]
Similar looking one from Jan 2:

Log | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/638/console
-- | --
Build Scans | https://gradle-enterprise.elastic.co/s/sm6wmy73dyhuq

And also one from 7.x branch:


Log | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk13,ES_RUNTIME_JAVA=openjdk13,nodes=general-purpose/417/console
-- | --
Build Scans | [7.5.2] https://gradle-enterprise.elastic.co/s/suphrrrj2zhyq [6.8.7] https://gradle-enterprise.elastic.co/s/xmnr2wdzqfqmmhttps://gradle-enterprise.elastic.co/s/d27o5sif3olkm

I've seen two more similar looking ones going back until Dec 20th, this seems to be the earlies one I can find atm:


Log | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk13,ES_RUNTIME_JAVA=java8,nodes=general-purpose/400/console
-- | --
Build Scans | [7.5.2] https://gradle-enterprise.elastic.co/s/s7ooeu6pz7myk [6.8.7] https://gradle-enterprise.elastic.co/s/zqshf7y26pvjghttps://gradle-enterprise.elastic.co/s/r2thietkm2vpc

**Comment by tbrooks8** [comment]
Should be fixed by #50620
